src/poetry/locations.py

```diff
@@ -4,21 +4,20 @@
 
 from pathlib import Path
 
-from poetry.utils.appdirs import user_cache_dir
-from poetry.utils.appdirs import user_config_dir
-from poetry.utils.appdirs import user_data_dir
+from platformdirs import user_cache_path
+from platformdirs import user_config_path
+from platformdirs import user_data_path
 
 
-CACHE_DIR = user_cache_dir("pypoetry")
-DATA_DIR = user_data_dir("pypoetry")
-CONFIG_DIR = user_config_dir("pypoetry")
+CACHE_DIR = user_cache_path("pypoetry", appauthor=False)
+CONFIG_DIR = user_config_path("pypoetry", appauthor=False, roaming=True)
```

Comment:
For this release we might want to issue a deprecation log for osx users and suggest user moves the directory if it already exists.

Something like:

```py
logger = logging.getLogger(__name__)

if osx:
    _LEGACY_PATH = CONFIG_DIR.parent.parent.joinpath("Application Support").joinpath("pypoetry")
    
    if _LEGACY_PATH.exists():
        logger.warn("Legacy configuration directory exists at %s, reusing this directory. Consider moving this to %s, as support for legacy directory will be removed in an upcoming release.", _LEGACY_PATH, CONFIG_DIR)
        CONFIG_DIR = _LEGACY_PATH
```

Reply:
Seems reasonable.  I've pushed a commit that's actually a bit more careful even than that, checking explicitly for `config.toml` or `auth.toml`.  

The reason is that `../ApplicationSupport/pypoetry` remains the correct directory for application data - and while `DATA_DIR` seems unused (and is removed in this MR) I'm cautious of the possibility that it might exist anyway, for non-config purposes..